worlds/oot_soh/location_access/overworld/market.py

```diff
@@ -5,8 +5,272 @@
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_SPOOKY_MASK, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_BUNNY_HOOD, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_RIGHT_MASKS, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+    ])
+    # Locations
+    add_locations(Regions.MARKET_MASK_SHOP, world, [
+        (Locations.MARKET_MASK_SHOP_HINT, lambda bundle: True)
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BAZAAR, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Shooting Gallery
+    # Locations
+    add_locations(Regions.MARKET_SHOOTING_GALLERY, world, [
+        (Locations.MARKET_SHOOTING_GALLERY, lambda bundle: (is_child(bundle) and has_item(Items.CHILD_WALLET, bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_SHOOTING_GALLERY, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Bombchu Bowling
+    # Events
+    add_events(Regions.MARKET_BOMBCHU_BOWLING, world, [
+        (EventLocations.MARKET_BOMBCHU_BOWLING, LocalEvents.CAN_PLAY_BOWLING, lambda bundle: (has_item(Items.CHILD_WALLET, bundle)))
+    ])
+    # Locations
+    add_locations(Regions.MARKET_BOMBCHU_BOWLING, world, [
+        (Locations.MARKET_BOMBCHU_BOWLING_FIRST_PRIZE, lambda bundle: (LocalEvents.CAN_PLAY_BOWLING and bombchus_enabled(bundle)))
+        (Locations.MARKET_BOMBCHU_BOWLING_SECOND_PRIZE, lambda bundle: (LocalEvents.CAN_PLAY_BOWLING and bombchus_enabled(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BOMBCHU_BOWLING, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Potion Shop
+    # Locations
+    add_locations(Regions.MARKET_POTION_SHOP, world, [
+        (Locations.MARKET_POTION_SHOP_ITEM1, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM2, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM3, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM4, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM5, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM6, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM7, lambda bundle: True),
+        (Locations.MARKET_POTION_SHOP_ITEM8, lambda bundle: True),
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_POTION_SHOP, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Treasure Chest Game
+    # Locations
+    add_locations(Regions.MARKET_TREASURE_CHEST_GAME, world, [
+        (Locations.MARKET_GREG_HINT, lambda bundle: (has_item(Items.CHILD_WALLET, bundle))),
+        # Todo: Missing 'shuffle_chest_minigame' option. Are we also missing 'logic->SmallKeys(SCENE_TREASURE_BOX_SHOP)
```

Comment:
You can remove the hint location here too, and completely ignore anything about the treasure chest minigame rule except for `logic->HasItem(RG_CHILD_WALLET) && ((logic->CanUse(RG_LENS_OF_TRUTH)`. It's all code that prepares for an eventual treasure chest minigame shuffle that hasn't been made in Ship yet, and will likely change when it is implemented properly anyway.